### PR TITLE
PR: Add inverse (arc) secant

### LINKED ISSUE #231: RFC: add support for computing the inverse/arc secant (asec)
## Checklist

> Please ensure the following tasks are completed before submitting a feature request.

-   [x] Read and understood the [Code of Conduct][code-of-conduct].
-   [x] Searched for existing issues and pull requests.
-   [x] The issue name begins with `RFC:`.

## Description

> Description of the feature request.

This RFC proposes adding support for computing the inverse (arc) secant. The initial implementation is straightforward:

```javascript
return acos( 1.0 / x );
```

Package: `@stdlib/math/base/special/asec`
Alias: `base.asec`

## Related Issues

> Does this feature request have any related issues?

No.

## Questions

> Any questions for reviewers?

No.

## Other

> Any other information relevant to this feature request? This may include screenshots, references, sample output, and/or implementation notes.

Reference implementations:

-   [Julia](https://github.com/JuliaLang/julia/blob/dee7d77c199c4776cf4e06afb7ce920f09ad3c8c/base/special/trig.jl#L983)

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by steff456**
Resolves #231

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds support to compute the inverse/arc secant

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   fixes #231 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `-0`', function test( t ) {
+	var v = asec( -0.0 );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` on the interval `]-1, 1[`', function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.js]: Should the interval be `[-1, 1]` here?

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Looks good, just had one comment concerning an interval in a test description.

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.js (hunk 1 above)
Comment (steff456) [reply lib/node_modules/@stdlib/math/base/special/asec/test/test.js]: The values for -1 and 1 are defined, so it is not inclusive

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/asec/test/test.js]: If not inclusive, then would open-interval notation be better? i.e., `(-1, 1)`.

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/asec/test/test.js]: Yes, I believe that has been our convention.

Comment (steff456) [reply lib/node_modules/@stdlib/math/base/special/asec/test/test.js]: I'll change it! I wasn't aware about the convention 😅

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/asec/test/test.js]: No worries, thanks for changing!

File: lib/node_modules/@stdlib/math/base/special/asec/benchmark/benchmark.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+// MAIN //
+
+bench( pkg, function benchmark( b ) {
+	var y;
+	var i;
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		y = asec( randu() );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asec/benchmark/benchmark.js]: This should fail atm. Instead, probably want something like
```suggestion
		y = asec( ( randu()*1000.0 ) + 1.0 );
```

File: lib/node_modules/@stdlib/math/base/special/asec/examples/index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+var linspace = require( '@stdlib/array/base/linspace' );
+var asec = require( './../lib' );
+
+var x = linspace( 0.01, 1.0, 100 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asec/examples/index.js]: The value range should be updated. This should return all `NaN` values atm.

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/asec/examples/index.js]: In the README, the value range is `[1.0, 10.0]`.

File: lib/node_modules/@stdlib/math/base/special/asec/lib/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+*
+* @module @stdlib/math/base/special/asec
+*
+* @example
+* var asec = require( '@stdlib/math/base/special/asec' );
+*
+* var v = asec( 1.0 );
+* // returns 0.0
+*
+* v = asec( 2 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asec/lib/index.js]: Missing decimal.

File: lib/node_modules/@stdlib/math/base/special/asec/docs/repl.txt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    Returns
+    -------
+    y: number
+        Inverse (arc) secant.
+
+    Examples
+    --------
+    > var y = {{alias}}( 1.0 )
+    0.0
+    > y = {{alias}}( 2 )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asec/docs/repl.txt]: Missing decimal.

File: lib/node_modules/@stdlib/math/base/special/asec/lib/index.js (hunk 4 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/asec/lib/index.js]: @steff456 This is undoubtedly not obvious, but, for the most part, when an input value is a floating-point number, we always include decimals, similar to Python. When a function only takes integer values, we omit the decimal to indicate that only integer inputs are to be provided.

File: lib/node_modules/@stdlib/math/base/special/asec/lib/main.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+*
+* @param {number} x - input value
+* @returns {number} inverse (arc) secant
+*
+* @example
+* var v = asec( 1.0 );
+* // returns 0.0
+*
+* @example
+* var v = asec( 2 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asec/lib/main.js]: Missing decimal.

File: lib/node_modules/@stdlib/math/base/special/asec/package.json [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      "url": "https://github.com/stdlib-js/stdlib/graphs/contributors"
+    }
+  ],
+  "main": "./lib",
+  "directories": {
+    "benchmark": "./benchmark",
+    "doc": "./docs",
+    "example": "./examples",
+    "lib": "./lib",
+    "scripts": "./scripts",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asec/package.json]: This package does not appear to have a `scripts` folder. This entry can be removed.

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.js [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+
+tape( 'the function returns `NaN` if provided `NaN`', function test( t ) {
+	var v = asec( NaN );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `+0`', function test( t ) {
+	var v = asec( 0.0 );
+	t.equal( isnan(v), true, 'returns NaN' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.js]:
```suggestion
	t.equal( isnan( v ), true, 'returns NaN' );
```

To be consistent with L244.

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.js [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` on the interval `(-1, 1)`', function test( t ) {
+	var v = asec( 0.5 );
+	var w = asec( -0.5 );
+
+	t.equal( isnan(v), true, 'returns NaN' );
+	t.equal( isnan(w), true, 'returns NaN' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.js]:
```suggestion
	t.equal( isnan( v ), true, 'returns NaN' );
	t.equal( isnan( w ), true, 'returns NaN' );
```

To be consistent with above.

Typically, we include spaces around arguments, with exceptions for general readability when there are many nested parentheses.

**Review by kgryte — COMMENTED** [review]
For the most part, looks good! Thanks, @steff456!

Just a few nitpicks, but otherwise, should be good to go after making the changes and another quick review!

File: lib/node_modules/@stdlib/math/base/special/asec/lib/index.js (hunk 4 above)
Comment (steff456) [reply lib/node_modules/@stdlib/math/base/special/asec/lib/index.js]: I think there's some other packages that need to be updated for consistency, I copied from `asech` and it is missing decimal places also 😉

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/asec/lib/index.js]: Feel free to flag those instances so we can fix them. :)

File: lib/node_modules/@stdlib/math/base/special/asec/test/fixtures/julia/runner.jl [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+# Examples
+
+``` julia
+julia> x = range( -1000, stop = 1000, length = 2022 );
+julia> gen( x, \"data.json\" );
+```
+"""
+function gen( domain, name )
+	x = collect( domain );
+	y = acos.( 1 ./ x );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asec/test/fixtures/julia/runner.jl]:
```suggestion
	y = acos.( 1.0 ./ x );
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.js (hunk 8 above)
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.js]:
```suggestion
	t.equal( isnan( v ), true, 'returns NaN' );
```

**Review by kgryte — APPROVED** [review]
LGTM! Thanks, @steff456! And thanks, @Planeshifter for the initial review!

**Pull request merged**